Tests.Unit.Core.ApplicationServices/ApplicationServices/Organizations/StsOrganizationSynchronizationServiceTest.cs

```diff
@@ -557,6 +570,102 @@ public void Disconnect_Succeeds_And_Converts_All_Imported_Org_Units_To_Kitos_Uni
         }
 
+        [Fact]
+        public void GetChangeLogForOrganization_Returns_All_Logs()
+        {
+            var orgUuid = A<Guid>();
+
+            var logs = new List<StsOrganizationConsequenceLog> { new (), new () };
+            var changeLogs = new List<StsOrganizationChangeLog> { new () { ConsequenceLogs = logs }, new() { ConsequenceLogs = logs } };
+
+            var stsOrganizationConnection = new StsOrganizationConnection
+            {
+                Connected = true,
+                SynchronizationDepth = A<int>(),
+                StsOrganizationChangeLogs = changeLogs
+            };
+            var organization = new Organization
+            {
+                Uuid = orgUuid,
+                StsOrganizationConnection = stsOrganizationConnection
+            };
+            
+            SetupGetOrganizationReturns(orgUuid, organization);
+
+            var result = _sut.GetChangeLogForOrganization(orgUuid);
+
+            Assert.True(result.Ok);
+            Assert.Equal(2, result.Value.Count());
+        }
+
+        [Fact]
+        public void GetChangeLogForOrganization_Returns_Number_Of_Logs()
+        {
+            var orgUuid = A<Guid>();
+
+            var logs = new List<StsOrganizationConsequenceLog> { new (), new () };
+            var changeLogs = new List<StsOrganizationChangeLog> { new () { ConsequenceLogs = logs }, new() { ConsequenceLogs = logs } };
+
+            var stsOrganizationConnection = new StsOrganizationConnection
+            {
+                Connected = true,
+                SynchronizationDepth = A<int>(),
+                StsOrganizationChangeLogs = changeLogs
+            };
+            var organization = new Organization
+            {
+                Uuid = orgUuid,
+                StsOrganizationConnection = stsOrganizationConnection
+            };
+
+
+            SetupGetOrganizationReturns(orgUuid, organization);
+
+            var result = _sut.GetChangeLogForOrganization(orgUuid, 1);
+
+            Assert.True(result.Ok);
+            var logResult = Assert.Single(result.Value);
+            ;
+            Assert.Equal(2, logResult.ConsequenceLogs.Count);
+        }
+
+        [Fact]
+        public void GetChangeLogForOrganization_GetOrganization_Returns_OperationError()
+        {
+            var orgUuid = A<Guid>();
+
+            SetupGetOrganizationReturns(orgUuid, new OperationError(OperationFailure.NotFound));
+
+            var result = _sut.GetChangeLogForOrganization(orgUuid, 1);
+
+            Assert.True(result.Failed);
+            Assert.Equal(OperationFailure.NotFound, result.Error);
+        }
+
+        [Fact]
+        public void GetChangeLogForOrganization_GetOrganization_Returns_BadInput()
+        {
+            var orgUuid = A<Guid>();
+            
+            var stsOrganizationConnection = new StsOrganizationConnection
+            {
+                Connected = true,
```

Comment:
what about the test that proves that the logs are written when an update is performed?